var/spack/repos/builtin/packages/motif/package.py

```diff
@@ -37,6 +37,11 @@ class Motif(AutotoolsPackage):
 
     patch('add_xbitmaps_dependency.patch')
 
+    # Solve "implicit declaration of function is invalid in C99 error" with apple-clang:
+    def setup_build_environment(self, spack_env):
+        if spec.satisfies('%apple-clang'):
+            spack_env.set('CFLAGS', '-Wno-implicit-function-declaration')
```

Comment:
:thinking: Can we solve this with a patch that fixes the code rather than ignoring the error?

Reply:
Possibly. I felt like the actual fix is to use `-std=c11` but that didn't work for me. I will take another look at this when I get a chance.

Reply:
Long time since I last coded in C, but I think the error here is saying that you have an implicit declaration somewhere. This is usually due to a header (with the explicit declaration of the function signature) that was not included in the corresponding source file and the flag is just suppressing that warning. Implicit declaration is legal in C89, but was declared illegal in later standards. See for instance [here](https://stackoverflow.com/questions/17508503/is-implicit-function-declaration-legal-in-c89)

Reply:
@jrood-nrel